src/codeu/chat/client/maingui/MainGui.java

```diff
@@ -53,25 +57,79 @@ private void initialize() {
     mainFrame = new JFrame("Chat");
     mainFrame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
     mainFrame.setSize(790, 450);
-    //mainFrame.setPreferredSize(new Dimension(790,450));
-
-    // Main View - outermost graphics panel.
-    final JPanel mainViewPanel = new JPanel(new BorderLayout());
 
-    // Build login panel
-    final JPanel loginViewPanel = new LoginPanel(clientContext,this);
+     // Main View - outermost graphics panel.
+    mainViewPanel = new JPanel(new BorderLayout());
 
-    mainViewPanel.add(loginViewPanel,BorderLayout.CENTER);
+    // Check if user is logged in order to open the LoginPanel or the other panels.
+    if(!clientContext.user.hasCurrent()){
+      loadLoginPanel();
+    }else{
+      loadChatPanels();
+    }
 
     mainFrame.add(mainViewPanel);
     mainFrame.pack();
   }
+  public void loadLoginPanel(){
+    final LoginPanel loginViewPanel = new LoginPanel(clientContext,this);
+
+    mainViewPanel.add(loginViewPanel,BorderLayout.CENTER);
+    mainFrame.pack();
+  }
   
+  public void loadChatPanels(){
+    final MessagePanel messageViewPanel = new MessagePanel(clientContext);
+    final ConversationPanel conversationsPanel = new ConversationPanel(clientContext, messageViewPanel);
+    final MenuBarPanel menuBarPanel = new MenuBarPanel(clientContext,this);
+
+    mainViewPanel.add(messageViewPanel,BorderLayout.CENTER);
+    mainViewPanel.add(conversationsPanel,BorderLayout.WEST);
+    mainViewPanel.add(menuBarPanel,BorderLayout.NORTH);
+    mainFrame.pack();
+    
+  }
   // Login request callback function 
   @Override
   public void onLoginRequest(String username, String pswd){
     
-    // Login function
-    JOptionPane.showMessageDialog(mainFrame,"User: " + username + "\nPswd: " + pswd);
+    // Search for user in the server's database
+    final User loginUser = clientContext.user.checkUserInDatabase(username, pswd);
+
+    if(loginUser != null){
```

Comment:
pick a consistent style wrt to spaces before/after parenthesis.

Elsewhere you've written this like:
if (loginUser != null) {